src/extensions/http_api/client.js

```diff
@@ -213,7 +213,8 @@ class HttpApiClient {
                 method,
                 qs: this.query || {},
                 headers: this.headers,
-                jar: this.cookieJar
+                jar: this.cookieJar,
+                followRedirect: this.headers.followRedirect !== 'false'
```

Comment:
The option should not be defined through headers, but via a step definition like:

```
/^(?:I )?follow redirect$/
```

and correspond to a `followRedirect` option of the client, it should be reset like all other client's parameters.